x-pack/solutions/security/plugins/security_solution/public/flyout/attack_details/hooks/use_tabs.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useMemo } from 'react';
+import type { PanelPath } from '@kbn/expandable-flyout';
+import type { AttackDetailsPanelPaths } from '..';
+
+import * as tabs from '../tabs';
+import type { AttackDetailsPanelTabType } from '../tabs';
+import { useKibana } from '../../../common/lib/kibana';
+import { FLYOUT_STORAGE_KEYS } from '../constants/local_storage';
+
+export const tabsDisplayed = [tabs.overviewTab, tabs.tableTab, tabs.jsonTab];
+
+export interface UseTabsParams {
+  /**
+   * The path passed in when using the expandable flyout API to open a panel.
+   */
+  path: PanelPath | undefined;
+}
+
+export interface UseTabsResult {
+  /**
+   * The tabs to display in the attack details panel.
+   */
+  tabsDisplayed: AttackDetailsPanelTabType[];
+  /**
+   * The tab id to selected in the attack details panel.
+   */
+  selectedTabId: AttackDetailsPanelPaths;
+}
+
+/**
+ * Hook to get the tabs to display in the attack details panel and the selected tab.
+ */
+export const useTabs = ({ path }: UseTabsParams): UseTabsResult => {
+  const { storage } = useKibana().services;
+
+  const selectedTabId = useMemo(() => {
+    // we use the value passed from the url and use it if it exists in the list of tabs to display
+    if (path) {
+      const selectedTab = tabsDisplayed.map((tab) => tab.id).find((tabId) => tabId === path.tab);
+      if (selectedTab) {
+        return selectedTab;
+      }
+    }
+
+    // we check the tab saved in local storage and use it if it exists in the list of tabs to display
+    const tabSavedInlocalStorage = storage.get(FLYOUT_STORAGE_KEYS.RIGHT_PANEL_SELECTED_TABS);
+    if (
+      tabSavedInlocalStorage &&
```

Comment:
Consider moving this condition outside of the `if` statement to avoid multiple operations there. Something like 
```
const savedTabExists = tabSavedInlocalStorage && tabsDisplayed.map((tab) => tab.id).includes(tabSavedInlocalStorage);
if (savedTabExists) {
  return tabSavedInlocalStorage;
}
```